### PR TITLE
LPS-110936 SF

### PR DESCRIPTION
**Pull request by CarlosBrichete**

### CODE REVIEW

**Comment by CarlosBrichete** [comment]
ci:test:sf

**Comment by CarlosBrichete** [comment]
ci:test:relevant

File: portal-web/test/functional/com/liferay/portalweb/macros/ABTesting.macro [hunk 1]
```diff
@@ -0,0 +1,55 @@
+definition {
```
Comment (manoelcyreno) [inline portal-web/test/functional/com/liferay/portalweb/macros/ABTesting.macro]: change the name to "ABTest.macro"

File: portal-web/test/functional/com/liferay/portalweb/macros/AnalyticsCloud.macro [hunk 2]
```diff
@@ -0,0 +1,66 @@
+definition {
+
+	macro loginAC {
```
Comment (manoelcyreno) [inline portal-web/test/functional/com/liferay/portalweb/macros/AnalyticsCloud.macro:3]: please, not use acronym...

File: portal-web/test/functional/com/liferay/portalweb/macros/AnalyticsCloud.macro [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+		Click(locator1 = "AnalyticsCloudConnection#ANALYTICS_CLOUD_PASSWORD_LOGIN_FIELD");
+
+		Type(
+			locator1 = "AnalyticsCloudConnection#ANALYTICS_CLOUD_PASSWORD_LOGIN_FIELD",
+			value1 = "${Password}");
+
+		Click(locator1 = "AnalyticsCloudConnection#ANALYTICS_CLOUD_LOGIN_BUTTON");
+	}
+
+	macro syncAC {
```
Comment (manoelcyreno) [inline portal-web/test/functional/com/liferay/portalweb/macros/AnalyticsCloud.macro:21]: please, not use acronym...

File: portal-web/test/functional/com/liferay/portalweb/paths/portlet/abtest/ABTest.path [hunk 4]
```diff
@@ -68,6 +68,21 @@
 	<td>//*[contains(@class,'btn btn-primary') and contains(text(),'Save')]</td>
 	<td></td>
 </tr>
+<tr>
+	<td>VARIANT_DROPDOWN</td>
+	<td>//div[@class='autofit-col']//*[contains(@class, 'dropdown-toggle btn-monospaced btn btn-outline-borderless btn-outline-secondary')]</td>
```
Comment (manoelcyreno) [inline portal-web/test/functional/com/liferay/portalweb/paths/portlet/abtest/ABTest.path:73]: here you need to use all of these classes?

File: portal-web/test/functional/com/liferay/portalweb/paths/portlet/analyticscloudconnection/AnalyticsCloudConnection.path [hunk 5]
```diff
@@ -43,6 +43,21 @@
 	<td>//*[contains(@type, 'checkbox')]</td>
 	<td></td>
 </tr>
+<tr>
+	<td>INSIDE_DONE_BUTTON</td>
+	<td>//*[contains(@id, '_com_liferay_configuration_admin_web_portlet_InstanceSettingsPortlet_svzz')]</td>
```
Comment (manoelcyreno) [inline portal-web/test/functional/com/liferay/portalweb/paths/portlet/analyticscloudconnection/AnalyticsCloudConnection.path:48]: svzz is a random data

File: portal-web/test/functional/com/liferay/portalweb/paths/portlet/abtest/ABTest.path (hunk 4 above)
Comment (CarlosBrichete) [reply portal-web/test/functional/com/liferay/portalweb/paths/portlet/abtest/ABTest.path:73]: yes, it was the only way i've found to find the path...

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/abtest/CreateABTest.testcase [hunk 6]
```diff
@@ -24,22 +24,7 @@ definition {
...
-
-			Click(locator1 = "AnalyticsCloudConnection#SYNCED_SITES_BUTTON");
-
-			Click(locator1 = "AnalyticsCloudConnection#ADD_NEW_PROPERTY_BUTTON");
-
-			Click(locator1 = "AnalyticsCloudConnection#INPUT_LIFERAY_DXP_CHECKBOX");
-
-			Click(locator1 = "AnalyticsCloudConnection#DONE_BUTTON");
-		}
+		AnalyticsCloud.syncAC();
```
Comment (manoelcyreno) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/abtest/CreateABTest.testcase:27]: please, not use acronym...

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/abtest/CreateABTest.testcase [hunk 7]
```diff
@@ -55,66 +40,49 @@ definition {
 	@description = "Validate if the user can create an A/B Test"
 	@priority = "5"
 	test CreateABTestByBounceRate {
+		ABTesting.createABtestByBounceRate(TestTitle = "AB test title");
```
Comment (manoelcyreno) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/abtest/CreateABTest.testcase]: rename ABtest to ABTest

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/abtest/CreateABTest.testcase [hunk 8]
```diff
@@ -55,66 +40,49 @@ definition {
...
 
-			Click(locator1 = "ABTest#SAVE_VARIANT_NAME_BUTTON");
-		}
-
-		task ("Run the A/B Test") {
-			Click(locator1 = "ABTest#REVIEW_AND_RUN_TEST_BUTTON");
+		ABTesting.runTest();
 
-			Click(locator1 = "ABTest#RUN_TEST_BUTTON");
+		ABTesting.checkAC();
```
Comment (manoelcyreno) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/abtest/CreateABTest.testcase]: please, not use acronym...

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/abtest/CreateABTest.testcase [hunk 9]
```diff
@@ -55,66 +40,49 @@ definition {
...
 
-			Click(locator1 = "ABTest#OK_BUTTON");
-		}
+		AnalyticsCloud.loginAC(
```
Comment (manoelcyreno) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/abtest/CreateABTest.testcase:51]: please, not use acronym...

**Comment by manoelcyreno** [comment]
rename all your variables to start with the lowerCase...

**Pull request closed without merging**